app/src/main/kotlin/com/wire/android/ui/authentication/create/common/CreateAccountBaseViewModel.kt

```diff
@@ -193,17 +197,37 @@ abstract class CreateAccountBaseViewModel(
                         email = emailState.email.text.trim(),
                         emailActivationCode = codeState.code.text.text
                     )
-                CreateAccountFlowType.CreateTeam -> TODO()
+                CreateAccountFlowType.CreateTeam ->
+                    RegisterParam.Team(
+                        firstName = detailsState.firstName.text.trim(),
+                        lastName = detailsState.lastName.text.trim(),
+                        password = detailsState.password.text,
+                        email = emailState.email.text.trim(),
+                        emailActivationCode = codeState.code.text.text,
+                        teamName = detailsState.teamName.text.trim(),
+                        teamIcon = "default"
+                    )
             }
 
-            val codeError = registerAccountUseCase(registerParam, serverConfig)
-                .let {
-                    if (it is RegisterResult.Success) // TODO what if user creates an account but doesn't register a new device?
-                        clientScopeProviderFactory.create(it.value.second.userId).clientScope.register(
-                            password = registerParam.password,
-                            capabilities = null
-                        ).toCodeError()
-                    else it.toCodeError()
+            val codeError = registerAccountUseCase(
```

Comment:
are you consuming those somewhere in the UI ? :P it starts to look like MVVM more I think than MVI :p

Reply:
Yes, it's a part of CodeState and it's used on CodeScreen. 😉